test/execution/test_protocol.py

```diff
@@ -305,3 +175,41 @@ def protocol(role1, role2):
 
     assert protocol.role_assignments.assignments["role1"] == [bob]
     assert protocol.role_assignments.assignments["role2"] == [alice]
+
+
+def test_execution_with_communications(workers):
+    @sy.func2protocol(roles=["role1", "role2"])
+    def protocol(role1, role2):
+        x = role1.torch.rand([1])
+        y = role1.torch.rand([1])
+
+        z = role2.torch.rand([1])
+
+        # On role1
+        a = x + y
+        b = a.send(role2)
+
+        # On role2
+        res = b + z
+        # d = c.send(role1)
+
+        # # On role1
+        # res = d * 2
+
```

Comment:
Are these commented lines working at this point? If not, we can probably just remove them.

Reply:
Oops, yeah, we can remove them! I thought I could have back and forth between workers directly but this needs a bit more work. We simply execute all the actions for all the `Roles` sequentially on this branch

Reply:
Ah, okay, cool! Could you create an issue for the further work then?